vaadin-select-flow/src/main/java/com/vaadin/flow/component/select/Select.java

```diff
@@ -755,8 +765,22 @@ private void reset() {
         if (isEmptySelectionAllowed()) {
             addEmptySelectionItem();
         }
-        getDataProvider().fetch(new Query<>()).map(this::createItem)
-                .forEach(this::add);
+
+        synchronized (dataProvider) {
+            final AtomicInteger itemCounter = new AtomicInteger(0);
+            getDataProvider().fetch(new Query<>()).map(this::createItem)
+                    .forEach(component -> {
+                        add(component);
+                        itemCounter.incrementAndGet();
+                    });
+            lastFetchedDataSize = itemCounter.get();
+        }
+
+        runBeforeClientResponse(ui -> {
```

Comment:
Small thing. I guess we should have the fireSizeEvent stored the first time into a `SerializableConsumer<UI> sizeRequeest` so that we don't register multiple beforeClientResponses.
so something like 
```java
if(sizeRequest == null) {
  sizeRequest = context -> [
    fireSizeEvent();
    sizeRequest = null;
  }
  runBeforeClientResponse(ui ->sizeRequest());
}

}
```

Reply:
Does that mean the most recent resets (and thus data fetch requests) would be ignored unless the first sizeRequest is executed? Won't we need the most recent of them?

Reply:
it will run just before the client response so it will always be for the last fired event.